spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTopicTests.java

```diff
@@ -0,0 +1,52 @@
+package org.springframework.pulsar.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.util.stream.Stream;
+import org.apache.pulsar.common.naming.TopicDomain;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class PulsarTopicTests {
+
+	@Test
+	void testBuilderDefaultValues() {
+		String topicName = "test-default-values";
+		PulsarTopicBuilder builder = PulsarTopic.builder(topicName);
+		PulsarTopic topic = builder.build();
+
+		assertThat(topic.topicName()).isEqualTo(topicName);
+		assertThat(topic.numberOfPartitions()).isEqualTo(0);
+	}
+
+	@ParameterizedTest
+	@MethodSource("testTopicComponentsProvider")
+	void testTopicComponents(PulsarTopic topic, TopicDomain domain, String tenant, String namespace, String topicName) {
```

Comment:
**[TinyNit]** One of the cool things w/ Junit5 was no longer needing to name test cases w/ the prefix `test`. It seems minute but it does clean things up a bit. Do you mind removing the renaming the tests? I can also do this in follow-on commit. Not trying to nit-pick you to death 😸